deepchecks/tabular/context.py

```diff
@@ -256,24 +272,27 @@ def model(self) -> BasicModel:
             self._validated_model = True
         return self._model
 
+    @property
+    def classes(self) -> t.List:
+        """Return ordered list of possible label classes for classification tasks or None for regression."""
+        return self._classes
+
     @property
     def model_name(self):
         """Return model name."""
         return type(self.model).__name__
 
     @property
     def task_type(self) -> TaskType:
-        """Return task type if model & train & label exists. otherwise, raise error."""
-        if self._task_type is None:
-            self._task_type = task_type_check(self.model, self.train)
-        return self._task_type
-
-    @property
-    def features_importance(self) -> t.Optional[pd.Series]:
-        """Return feature importance, or None if not possible."""
-        # TODO: remove in future
-        get_logger().warning('"features_importance" property is deprecated use "feature_importance" instead')
-        return self.feature_importance
+        """Return task type based on calculated classes argument."""
+        if self._classes is None:
+            return TaskType.REGRESSION
+        elif len(self._classes) == 2:
+            return TaskType.BINARY
+        elif len(self._classes) > 2:
+            return TaskType.MULTICLASS
+        else:
+            raise DatasetValidationError(f'Found only one class in label column: {self._classes}.')
```

Comment:
The way to solve this (from a user standpoint) is to pass a list of label_classes to the dataset. Need to tell them about it here